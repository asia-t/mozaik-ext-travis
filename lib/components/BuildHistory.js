'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _ui = require('@mozaik/ui');

var _BuildHistoryItem = require('./BuildHistoryItem');

var _BuildHistoryItem2 = _interopRequireDefault(_BuildHistoryItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var BuildHistory = function (_Component) {
    _inherits(BuildHistory, _Component);

    function BuildHistory() {
        _classCallCheck(this, BuildHistory);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    BuildHistory.getApiRequest = function getApiRequest(_ref) {
        var owner = _ref.owner,
            repository = _ref.repository,
            _ref$limit = _ref.limit,
            limit = _ref$limit === undefined ? BuildHistory.defaultProps.limit : _ref$limit;

        return {
            id: 'travis.repositoryBuildHistory.' + owner + '.' + repository + '.' + limit,
            params: { owner: owner, repository: repository, limit: limit }
        };
    };

    BuildHistory.prototype.render = function render() {
        var _props = this.props,
            owner = _props.owner,
            repository = _props.repository,
            title = _props.title,
            hideHeader = _props.hideHeader,
            apiData = _props.apiData,
            apiError = _props.apiError,
            theme = _props.theme;


        var body = _react2.default.createElement(_ui.WidgetLoader, null);
        var buildCount = void 0;
        if (apiData && !apiError) {
            buildCount = apiData.builds.pagination.count;
            body = _react2.default.createElement(
                'div',
                null,
                apiData.builds.items.map(function (build) {
                    return _react2.default.createElement(_BuildHistoryItem2.default, { key: build.id, build: build, theme: theme });
                })
            );
        }

        return _react2.default.createElement(
            _ui.Widget,
            null,
            !hideHeader && _react2.default.createElement(_ui.WidgetHeader, {
                title: title || 'builds',
                subject: title ? null : owner + '/' + repository,
                count: buildCount,
                icon: _ui.ActivityIcon
            }),
            _react2.default.createElement(
                _ui.WidgetBody,
                { disablePadding: true, isHeaderless: hideHeader },
                _react2.default.createElement(
                    _ui.TrapApiError,
                    { error: apiError },
                    body
                )
            )
        );
    };

    return BuildHistory;
}(_react.Component);

BuildHistory.propTypes = {
    owner: _propTypes2.default.string.isRequired,
    repository: _propTypes2.default.string.isRequired,
    title: _propTypes2.default.string,
    limit: _propTypes2.default.number.isRequired,
    hideHeader: _propTypes2.default.bool.isRequired,
    apiData: _propTypes2.default.shape({
        builds: _propTypes2.default.shape({
            pagination: _propTypes2.default.shape({
                count: _propTypes2.default.number.isRequired
            }).isRequired,
            items: _propTypes2.default.arrayOf(_BuildHistoryItem.BuildPropType).isRequired
        }).isRequired
    }),
    apiError: _propTypes2.default.object,
    theme: _propTypes2.default.object.isRequired
};
BuildHistory.defaultProps = {
    limit: 10,
    hideHeader: false
};
exports.default = BuildHistory;