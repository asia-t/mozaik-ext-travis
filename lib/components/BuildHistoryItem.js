'use strict';

exports.__esModule = true;
exports.BuildPropType = undefined;

var _templateObject = _taggedTemplateLiteralLoose(['\n    display: grid;\n    grid-template-columns: 0.6vmin auto;\n    background: ', ';\n    ', ' &:hover {\n        background: ', ';\n    }\n'], ['\n    display: grid;\n    grid-template-columns: 0.6vmin auto;\n    background: ', ';\n    ', ' &:hover {\n        background: ', ';\n    }\n']),
    _templateObject2 = _taggedTemplateLiteralLoose(['\n    padding: 1.2vmin 2vmin 1.8vmin 1.2vmin;\n    display: grid;\n    grid-template-columns: 2.6vmin 2fr 1fr;\n    grid-column-gap: 1vmin;\n    grid-row-gap: 0.2vmin;\n    align-items: center;\n'], ['\n    padding: 1.2vmin 2vmin 1.8vmin 1.2vmin;\n    display: grid;\n    grid-template-columns: 2.6vmin 2fr 1fr;\n    grid-column-gap: 1vmin;\n    grid-row-gap: 0.2vmin;\n    align-items: center;\n']),
    _templateObject3 = _taggedTemplateLiteralLoose(['\n    grid-column-start: 2;\n    grid-column-end: 4;\n    padding-top: 0.8vmin;\n    ', ';\n'], ['\n    grid-column-start: 2;\n    grid-column-end: 4;\n    padding-top: 0.8vmin;\n    ', ';\n']),
    _templateObject4 = _taggedTemplateLiteralLoose(['\n    color: ', ';\n'], ['\n    color: ', ';\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _ui = require('@mozaik/ui');

var _state = require('../lib/state');

var _duration = require('../lib/duration');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteralLoose(strings, raw) { strings.raw = raw; return strings; }

var BuildPropType = exports.BuildPropType = _propTypes2.default.shape({
    number: _propTypes2.default.string.isRequired,
    state: _propTypes2.default.string.isRequired,
    duration: _propTypes2.default.number.isRequired,
    finished_at: _propTypes2.default.string.isRequired,
    branch: _propTypes2.default.shape({
        name: _propTypes2.default.string.isRequired
    }).isRequired,
    commit: _propTypes2.default.shape({
        message: _propTypes2.default.string.isRequired
    }).isRequired
});

var Container = _styledComponents2.default.div(_templateObject, function (props) {
    return props.theme.list.item.background;
}, function (props) {
    return props.theme.list.item.extend.trim();
}, function (props) {
    return props.theme.list.item.hover.background;
});

var Content = _styledComponents2.default.div(_templateObject2);

var CommitMessage = _styledComponents2.default.div(_templateObject3, function (props) {
    return (0, _ui.typography)(props.theme, 'default', 'small');
});

var TextHighlight = _styledComponents2.default.span(_templateObject4, function (props) {
    return props.theme.colors.textHighlight;
});

var BuildHistoryItem = function (_Component) {
    _inherits(BuildHistoryItem, _Component);

    function BuildHistoryItem() {
        _classCallCheck(this, BuildHistoryItem);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    BuildHistoryItem.prototype.render = function render() {
        var _props = this.props,
            build = _props.build,
            theme = _props.theme;


        var color = (0, _state.colorByState)(theme.colors, build.state);
        var Icon = (0, _state.iconByState)(build.state);

        return _react2.default.createElement(
            Container,
            null,
            _react2.default.createElement('div', { style: { background: color } }),
            _react2.default.createElement(
                Content,
                null,
                _react2.default.createElement(Icon, { size: '2vmin', color: color }),
                _react2.default.createElement(
                    'span',
                    { style: { whiteSpace: 'pre' } },
                    _react2.default.createElement(
                        _ui.Text,
                        { variant: 'strong', style: { color: theme.colors.textHighlight } },
                        build.branch.name
                    ),
                    ' ',
                    _react2.default.createElement(_ui.GitCommitIcon, {
                        size: '1.6vmin',
                        style: { display: 'inline-block', verticalAlign: 'middle' }
                    }),
                    ' ',
                    _react2.default.createElement(
                        _ui.Text,
                        { variant: 'small' },
                        build.commit.sha.slice(0, 7)
                    )
                ),
                _react2.default.createElement(
                    'span',
                    { style: { whiteSpace: 'pre' } },
                    '#',
                    build.number,
                    ' ',
                    _react2.default.createElement(
                        _ui.Text,
                        { variant: 'small', style: { color: color } },
                        build.state
                    )
                ),
                _react2.default.createElement(
                    CommitMessage,
                    null,
                    build.commit.message
                ),
                build.finished_at && _react2.default.createElement(
                    _react.Fragment,
                    null,
                    _react2.default.createElement(
                        'span',
                        { style: { gridColumnStart: 2, gridColumnEnd: 4 } },
                        _react2.default.createElement(
                            _ui.Text,
                            { variant: 'small', style: { whiteSpace: 'pre' } },
                            _react2.default.createElement(_ui.ClockIcon, {
                                size: '1.6vmin',
                                style: { display: 'inline-block', verticalAlign: 'middle' }
                            }),
                            ' ',
                            'about',
                            ' ',
                            _react2.default.createElement(
                                TextHighlight,
                                null,
                                (0, _moment2.default)(build.started_at).fromNow()
                            ),
                            ' ',
                            'in',
                            ' ',
                            _react2.default.createElement(
                                TextHighlight,
                                null,
                                (0, _duration.secondsToString)(build.duration)
                            )
                        )
                    )
                )
            )
        );
    };

    return BuildHistoryItem;
}(_react.Component);

BuildHistoryItem.propTypes = {
    build: BuildPropType.isRequired,
    theme: _propTypes2.default.object.isRequired
};
exports.default = BuildHistoryItem;