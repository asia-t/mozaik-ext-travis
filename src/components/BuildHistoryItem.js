import React, { Fragment, Component } from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import styled from 'styled-components'
import { GitCommitIcon, ClockIcon, Text, typography } from '@mozaik/ui'
import { colorByState, iconByState } from '../lib/state'
import { secondsToString } from '../lib/duration'

export const BuildPropType = PropTypes.shape({
    number: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    duration: PropTypes.number.isRequired,
    finished_at: PropTypes.string.isRequired,
    branch: PropTypes.shape({
        name: PropTypes.string.isRequired,
    }).isRequired,
    commit: PropTypes.shape({
        message: PropTypes.string.isRequired,
    }).isRequired,
})

const Container = styled.div`
    display: grid;
    grid-template-columns: 0.6vmin auto;
    background: ${props => props.theme.list.item.background};
    ${props => props.theme.list.item.extend.trim()} &:hover {
        background: ${props => props.theme.list.item.hover.background};
    }
`

const Content = styled.div`
    padding: 1.2vmin 2vmin 1.8vmin 1.2vmin;
    display: grid;
    grid-template-columns: 2.6vmin 2fr 1fr;
    grid-column-gap: 1vmin;
    grid-row-gap: 0.2vmin;
    align-items: center;
`

const CommitMessage = styled.div`
    grid-column-start: 2;
    grid-column-end: 4;
    padding-top: 0.8vmin;
    ${props => typography(props.theme, 'default', 'small')};
`

const TextHighlight = styled.span`
    color: ${props => props.theme.colors.textHighlight};
`

class BuildHistoryItem extends Component {
    static propTypes = {
        build: BuildPropType.isRequired,
        theme: PropTypes.object.isRequired,
    }

    render() {
        const { build, theme } = this.props

        const color = colorByState(theme.colors, build.state)
        const Icon = iconByState(build.state)

        return (
            <Container>
                <div style={{ background: color }} />
                <Content>
                    <Icon size="2vmin" color={color} />
                    <span style={{ whiteSpace: 'pre' }}>
                        <Text variant="strong" style={{ color: theme.colors.textHighlight }}>
                            {build.branch.name}
                        </Text>{' '}
                        <GitCommitIcon
                            size="1.6vmin"
                            style={{ display: 'inline-block', verticalAlign: 'middle' }}
                        />{' '}
                        <Text variant="small">{build.commit.sha.slice(0, 7)}</Text>
                    </span>
                    <span style={{ whiteSpace: 'pre' }}>
                        #{build.number}{' '}
                        <Text variant="small" style={{ color }}>
                            {build.state}
                        </Text>
                    </span>
                    <CommitMessage>{build.commit.message}</CommitMessage>
                    {build.finished_at && (
                        <Fragment>
                            <span style={{ gridColumnStart: 2, gridColumnEnd: 4 }}>
                                <Text variant="small" style={{ whiteSpace: 'pre' }}>
                                    <ClockIcon
                                        size="1.6vmin"
                                        style={{ display: 'inline-block', verticalAlign: 'middle' }}
                                    />{' '}
                                    about{' '}
                                    <TextHighlight>
                                        {moment(build.started_at).fromNow()}
                                    </TextHighlight>{' '}
                                    in{' '}
                                    <TextHighlight>{secondsToString(build.duration)}</TextHighlight>
                                </Text>
                            </span>
                        </Fragment>
                    )}
                </Content>
            </Container>
        )
    }
}

export default BuildHistoryItem
