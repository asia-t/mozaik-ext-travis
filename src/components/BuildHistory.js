import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
    TrapApiError,
    Widget,
    WidgetHeader,
    WidgetBody,
    WidgetLoader,
    ActivityIcon,
} from '@mozaik/ui'
import BuildHistoryItem, { BuildPropType } from './BuildHistoryItem'

export default class BuildHistory extends Component {
    static propTypes = {
        owner: PropTypes.string.isRequired,
        repository: PropTypes.string.isRequired,
        title: PropTypes.string,
        limit: PropTypes.number.isRequired,
        hideHeader: PropTypes.bool.isRequired,
        apiData: PropTypes.shape({
            builds: PropTypes.shape({
                pagination: PropTypes.shape({
                    count: PropTypes.number.isRequired,
                }).isRequired,
                items: PropTypes.arrayOf(BuildPropType).isRequired,
            }).isRequired,
        }),
        apiError: PropTypes.object,
        theme: PropTypes.object.isRequired,
    }

    static defaultProps = {
        limit: 10,
        hideHeader: false,
    }

    static getApiRequest({ owner, repository, limit = BuildHistory.defaultProps.limit }) {
        return {
            id: `travis.repositoryBuildHistory.${owner}.${repository}.${limit}`,
            params: { owner, repository, limit },
        }
    }

    render() {
        const { owner, repository, title, hideHeader, apiData, apiError, theme } = this.props

        let body = <WidgetLoader />
        let buildCount
        if (apiData && !apiError) {
            buildCount = apiData.builds.pagination.count
            body = (
                <div>
                    {apiData.builds.items.map(build => (
                        <BuildHistoryItem key={build.id} build={build} theme={theme} />
                    ))}
                </div>
            )
        }

        return (
            <Widget>
                {!hideHeader && (
                    <WidgetHeader
                        title={title || 'builds'}
                        subject={title ? null : `${owner}/${repository}`}
                        count={buildCount}
                        icon={ActivityIcon}
                    />
                )}
                <WidgetBody disablePadding={true} isHeaderless={hideHeader}>
                    <TrapApiError error={apiError}>{body}</TrapApiError>
                </WidgetBody>
            </Widget>
        )
    }
}
