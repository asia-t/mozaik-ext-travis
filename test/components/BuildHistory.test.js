import React from 'react'
import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import renderer from 'react-test-renderer'
import { WidgetLoader, WidgetHeader, defaultTheme } from '@mozaik/ui'
import BuildHistory from '../../src/components/BuildHistory'
import BuildHistoryItem from '../../src/components/BuildHistoryItem'
import builds from '../fixtures/builds'

const owner = 'plouc'
const repository = 'mozaik'

jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }))

describe('getApiRequest', () => {
    it('should return correct api request', () => {
        expect(
            BuildHistory.getApiRequest({
                owner,
                repository,
            })
        ).toEqual({
            id: `travis.repositoryBuildHistory.${owner}.${repository}.10`,
            params: {
                limit: 10,
                owner,
                repository,
            },
        })
    })

    it('should support custom limit', () => {
        expect(
            BuildHistory.getApiRequest({
                owner,
                repository,
                limit: 35,
            })
        ).toEqual({
            id: `travis.repositoryBuildHistory.${owner}.${repository}.35`,
            params: {
                limit: 35,
                owner,
                repository,
            },
        })
    })
})

it('should display loader if no apiData available', () => {
    const wrapper = shallow(
        <BuildHistory owner={owner} repository={repository} theme={defaultTheme} />
    )

    expect(wrapper.find(WidgetLoader)).toHaveLength(1)
})

describe('title', () => {
    it('should display owner/repo by default', () => {
        const wrapper = shallow(
            <BuildHistory owner={owner} repository={repository} theme={defaultTheme} />
        )

        const header = wrapper.find(WidgetHeader)
        expect(header).toHaveLength(1)
        expect(header.prop('title')).toBe('builds')
        expect(header.prop('subject')).toEqual(`${owner}/${repository}`)
    })

    it('should allow title override', () => {
        const wrapper = shallow(
            <BuildHistory
                owner={owner}
                repository={repository}
                title="override"
                theme={defaultTheme}
            />
        )

        const header = wrapper.find(WidgetHeader)
        expect(header).toHaveLength(1)
        expect(header.prop('title')).toBe('override')
        expect(header.prop('subject')).toBe(null)
    })
})

it('should display builds info', () => {
    const wrapper = shallow(
        <BuildHistory
            owner={owner}
            repository={repository}
            theme={defaultTheme}
            apiData={{
                builds: {
                    items: builds,
                    pagination: { count: 10 },
                },
            }}
        />
    )

    const buildItems = wrapper.find(BuildHistoryItem)
    expect(buildItems).toHaveLength(builds.length)
})

it('should render as expected', () => {
    const tree = renderer.create(
        <ThemeProvider theme={defaultTheme}>
            <BuildHistory
                owner={owner}
                repository={repository}
                theme={defaultTheme}
                apiData={{
                    builds: {
                        items: builds,
                        pagination: { count: 10 },
                    },
                }}
            />
        </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
})
