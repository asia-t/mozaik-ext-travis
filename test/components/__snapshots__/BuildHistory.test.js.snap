// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div
  className="sc-bdVaJa eydJRb"
  style={Object {}}
>
  <div
    className="sc-bwzfXH bJUZVN"
  >
    <header
      className="sc-htpNat lcyfRu"
      style={Object {}}
    >
      <span>
        <div
          className="sc-ifAKCX rpFvk"
        >
          plouc/mozaik
        </div>
        builds
        <div
          className="sc-bxivhb eDwDKI"
        >
          10
        </div>
      </span>
      <div
        className="sc-EHOje iJdEwV"
      >
        <svg
          fill="none"
          height="2vmin"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="2vmin"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 12 18 12 15 21 9 3 6 12 2 12"
          />
        </svg>
      </div>
    </header>
    <div
      className="sc-bZQynM FCrnA"
      style={Object {}}
    >
      <div>
        <div
          className="sc-bRBYWo jxShFz"
        >
          <div
            style={
              Object {
                "background": "#8ddb8d",
              }
            }
          />
          <div
            className="sc-hzDkRC ghsfqN"
          >
            <svg
              fill="none"
              height="2vmin"
              stroke="#8ddb8d"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="2vmin"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              style={
                Object {
                  "whiteSpace": "pre",
                }
              }
            >
              <span
                className="sc-fMiknA boQYET"
                style={
                  Object {
                    "color": "#000",
                  }
                }
                type="default"
              >
                master
              </span>
               
              <svg
                fill="none"
                height="1.6vmin"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width="1.6vmin"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <line
                  x1="1.05"
                  x2="7"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="17.01"
                  x2="22.96"
                  y1="12"
                  y2="12"
                />
              </svg>
               
              <span
                className="sc-dVhcbM guLjQn"
                type="default"
              >
                d437772
              </span>
            </span>
            <span
              style={
                Object {
                  "whiteSpace": "pre",
                }
              }
            >
              #
              637
               
              <span
                className="sc-eqIVtm hCYmlC"
                style={
                  Object {
                    "color": "#8ddb8d",
                  }
                }
                type="default"
              >
                passed
              </span>
            </span>
            <div
              className="sc-jhAzac frhiXs"
            >
              feat(demo): upgrade demo dependencies
            </div>
            <span
              style={
                Object {
                  "gridColumnEnd": 4,
                  "gridColumnStart": 2,
                }
              }
            >
              <span
                className="sc-fAjcbJ efLRoA"
                style={
                  Object {
                    "whiteSpace": "pre",
                  }
                }
                type="default"
              >
                <svg
                  fill="none"
                  height="1.6vmin"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "display": "inline-block",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1.6vmin"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 6 12 12 16 14"
                  />
                </svg>
                 
                about
                 
                <span
                  className="sc-fBuWsC kdDwSw"
                >
                  2 days ago
                </span>
                 
                in
                 
                <span
                  className="sc-fBuWsC kdDwSw"
                >
                  5 min 27 sec
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          className="sc-bRBYWo jxShFz"
        >
          <div
            style={
              Object {
                "background": "#8ddb8d",
              }
            }
          />
          <div
            className="sc-hzDkRC ghsfqN"
          >
            <svg
              fill="none"
              height="2vmin"
              stroke="#8ddb8d"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="2vmin"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              style={
                Object {
                  "whiteSpace": "pre",
                }
              }
            >
              <span
                className="sc-caSCKo fPSyYT"
                style={
                  Object {
                    "color": "#000",
                  }
                }
                type="default"
              >
                @mozaik/ui@2.0.0-rc.2
              </span>
               
              <svg
                fill="none"
                height="1.6vmin"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width="1.6vmin"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <line
                  x1="1.05"
                  x2="7"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="17.01"
                  x2="22.96"
                  y1="12"
                  y2="12"
                />
              </svg>
               
              <span
                className="sc-gisBJw fLrGjf"
                type="default"
              >
                48bd4ba
              </span>
            </span>
            <span
              style={
                Object {
                  "whiteSpace": "pre",
                }
              }
            >
              #
              636
               
              <span
                className="sc-kjoXOD bVInpr"
                style={
                  Object {
                    "color": "#8ddb8d",
                  }
                }
                type="default"
              >
                passed
              </span>
            </span>
            <div
              className="sc-jhAzac frhiXs"
            >
              bump(@mozaik/ui): bump @mozaik/ui to version 2.0.0-rc.2
            </div>
            <span
              style={
                Object {
                  "gridColumnEnd": 4,
                  "gridColumnStart": 2,
                }
              }
            >
              <span
                className="sc-cHGsZl WCtRp"
                style={
                  Object {
                    "whiteSpace": "pre",
                  }
                }
                type="default"
              >
                <svg
                  fill="none"
                  height="1.6vmin"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "display": "inline-block",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1.6vmin"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 6 12 12 16 14"
                  />
                </svg>
                 
                about
                 
                <span
                  className="sc-fBuWsC kdDwSw"
                >
                  2 days ago
                </span>
                 
                in
                 
                <span
                  className="sc-fBuWsC kdDwSw"
                >
                  5 min 23 sec
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          className="sc-bRBYWo jxShFz"
        >
          <div
            style={
              Object {
                "background": "#8ddb8d",
              }
            }
          />
          <div
            className="sc-hzDkRC ghsfqN"
          >
            <svg
              fill="none"
              height="2vmin"
              stroke="#8ddb8d"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="2vmin"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              style={
                Object {
                  "whiteSpace": "pre",
                }
              }
            >
              <span
                className="sc-TOsTZ buljjU"
                style={
                  Object {
                    "color": "#000",
                  }
                }
                type="default"
              >
                master
              </span>
               
              <svg
                fill="none"
                height="1.6vmin"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width="1.6vmin"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <line
                  x1="1.05"
                  x2="7"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="17.01"
                  x2="22.96"
                  y1="12"
                  y2="12"
                />
              </svg>
               
              <span
                className="sc-kgAjT fCRwiO"
                type="default"
              >
                48bd4ba
              </span>
            </span>
            <span
              style={
                Object {
                  "whiteSpace": "pre",
                }
              }
            >
              #
              635
               
              <span
                className="sc-cJSrbW kPkduk"
                style={
                  Object {
                    "color": "#8ddb8d",
                  }
                }
                type="default"
              >
                passed
              </span>
            </span>
            <div
              className="sc-jhAzac frhiXs"
            >
              bump(@mozaik/ui): bump @mozaik/ui to version 2.0.0-rc.2
            </div>
            <span
              style={
                Object {
                  "gridColumnEnd": 4,
                  "gridColumnStart": 2,
                }
              }
            >
              <span
                className="sc-ksYbfQ cNsbJo"
                style={
                  Object {
                    "whiteSpace": "pre",
                  }
                }
                type="default"
              >
                <svg
                  fill="none"
                  height="1.6vmin"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "display": "inline-block",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1.6vmin"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <polyline
                    points="12 6 12 12 16 14"
                  />
                </svg>
                 
                about
                 
                <span
                  className="sc-fBuWsC kdDwSw"
                >
                  2 days ago
                </span>
                 
                in
                 
                <span
                  className="sc-fBuWsC kdDwSw"
                >
                  6 min 14 sec
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
